### PR TITLE
Pipe: implemented sdt-sampling-processor & refactored down-sampling-processor

### PR DESCRIPTION
**Pull request by xuanronaldo**

### CODE REVIEW

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/agent/plugin/dataregion/PipeDataRegionProcessorConstructor.java [hunk 1]
```diff
@@ -36,6 +37,10 @@ protected void initConstructors() {
     pluginConstructors.put(
         BuiltinPipePlugin.DO_NOTHING_PROCESSOR.getPipePluginName(), DoNothingProcessor::new);
     pluginConstructors.put(
-        BuiltinPipePlugin.DOWN_SAMPLING_PROCESSOR.getPipePluginName(), DownSamplingProcessor::new);
+        BuiltinPipePlugin.DOWN_SAMPLING_PROCESSOR.getPipePluginName(),
+        TumblingTimeSamplingProcessor::new);
+    pluginConstructors.put(
+        BuiltinPipePlugin.STD_FILTER_PROCESSOR.getPipePluginName(),
```
Comment (Caideyipi) [inline iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/agent/plugin/dataregion/PipeDataRegionProcessorConstructor.java:43]: SDT_FILTER

File: iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/processor/downsampling/PartialPathLastObjectCache.java [hunk 2]
```diff
@@ -30,21 +30,16 @@
...
 import org.slf4j.LoggerFactory;
 
-/**
- * Map-like component to look up for the last chosen time of a timeSeries. It has max size and
- * timeSeries may fail to find its last time and must design the logic to handle this.
- */
-public class PartialPathLastTimeCache implements AutoCloseable {
-
-  private static final Logger LOGGER = LoggerFactory.getLogger(PartialPathLastTimeCache.class);
+public class PartialPathLastObjectCache implements AutoCloseable {
```
Comment (Caideyipi) [inline iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/processor/downsampling/PartialPathLastObjectCache.java:33]: Consider re-writing the comments

File: iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/pipe/plugin/builtin/processor/downsampling/TumblingTimeProcessor.java [hunk 3]
```diff
@@ -27,4 +27,4 @@
  * pipe agent in the server module will replace this class with the real implementation when
  * initializing the Down Sampling processor.
  */
-public class DownSamplingProcessor extends PlaceHolderProcessor {}
+public class TumblingTimeProcessor extends PlaceHolderProcessor {}
```
Comment (Caideyipi) [inline iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/pipe/plugin/builtin/processor/downsampling/TumblingTimeProcessor.java:30]: Look at the "Down Sampling Processor" above

File: iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/pipe/plugin/builtin/processor/downsampling/SwingingDoorTrendingFilterProcessor.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iotdb.commons.pipe.plugin.builtin.processor.downsampling;
+
+import org.apache.iotdb.commons.pipe.plugin.builtin.processor.PlaceHolderProcessor;
+
+public class SwingingDoorTrendingFilterProcessor extends PlaceHolderProcessor {}
```
Comment (Caideyipi) [inline iotdb-core/node-commons/src/main/java/org/apache/iotdb/commons/pipe/plugin/builtin/processor/downsampling/SwingingDoorTrendingFilterProcessor.java:24]: Java docs?

**Comment by xuanronaldo** [comment]
self reviewed

**Review by SteveYurongSu — APPROVED** [review]
LGTM

**Comment by xuanronaldo** [comment]
LGTM

**Pull request merged**